team-list/src/main.js

```diff
@@ -0,0 +1,494 @@
+  }
+  var card = buildTeamListCard_(people);
+  return [card];
+}
+
+/**
+* Renders the contextual interface for a selected Drive file.
+*
+* @param {Object} event - current add-on event
+* @return {Card[]} Card(s) to display
+*/
+function onDriveItemsSelected(event) {
+  // For demo, only allow single select on files.
+  if (event.drive.selectedItems.length != 1) {
+    var message = 'To view team members collaborating on a file, select one file only.';
+    var card = buildSearchCard_(message);
+    return [card];
+  }
+
+  var selectedItem = event.drive.selectedItems[0];
+  if (!selectedItem.addonHasFileScopePermission) {
+    // Need file access to read ACL, ask user to authorize.
+    var authorizeFilesAction = CardService.newAction()
+        .setFunctionName('onAuthorizeDriveFiles')
+        .setLoadIndicator(CardService.LoadIndicator.SPINNER)
+        .setParameters({id: selectedItem.id});
+    var authMessageText = 'To view the people on your team the file is shared with, ' +
+                          'click *Authorize* to grant access.';
+    var authorizationMessage = CardService.newTextParagraph()
+        .setText(authMessageText);
+    var authorizeButton = CardService.newTextButton()
+        .setText('Authorize')
+        .setOnClickAction(authorizeFilesAction);
+    var card = CardService.newCardBuilder()
+        .addSection(CardService.newCardSection()
+            .addWidget(authorizationMessage)
+            .addWidget(authorizeButton))
+        .build();
+    return [card];
+  }
+
+  // Have access, extract ACLs to find co-workers
+  var emails = extractEmailsFromDrivePermissions_(event);
+  var people = fetchPeople_(emails);
+  if (people.length == 0) {
+    var card = buildSearchCard_('No team members found for current file.');
+    return [card];
+  }
+  var card = buildTeamListCard_(people);
+  return [card];
+}
+
+/**
+* Handles the click for requesting drive file access.
+*
+* @param {Object} event - current add-on event
+* @return {ActionResponse} Request to authorize access to a drive item
+*/
+function onAuthorizeDriveFiles(event) {
+  var id = event.parameters.id;
+  return CardService.newDriveItemsSelectedActionResponseBuilder()
+      .requestFileScope(id)
+      .build();
+}
+
+/**
+* Handles the user search request.
+*
+* @param {Object} event - current add-on event
+* @return {Card[]} Card(s) to display
+*/
+function onSearch(event) {
+  if (!event.formInputs || !event.formInputs.query) {
+    var notification = CardService.newNotification()
+        .setText('Enter a query before searching.');
+    return CardService.newActionResponseBuilder()
+        .setNotification(notification)
+        .build();
+  }
+
+  var query = event.formInputs.query[0];
+  var people = queryPeople_(query);
+
+  if (!people || people.length == 0) {
+    var notification = CardService.newNotification().setText('No people found.');
+    return CardService.newActionResponseBuilder()
+        .setNotification(notification)
```

Comment:
I see you're using notification here and the error case in buildSearchCard_ uses text paragraph but can you combine?

Reply:
Not without either making the code more complicated or introducing a helper method to build the notification (which adds complexity of its own, doesn't seem worth it for 2-3 lines of code.)